sys/net/network_layer/sixlowpan/lowpan.c

```diff
@@ -1110,7 +1110,7 @@ uint8_t lowpan_iphc_encoding(int if_id, const uint8_t *dest, int dest_len,
             else if (dest_len == 2 &&
                      ipv6_buf->destaddr.uint32[2] == HTONL(0x000000ff) &&
                      ipv6_buf->destaddr.uint16[6] == HTONS(0xfe00) &&
-                     ipv6_buf->destaddr.uint16[7] == *((uint16_t *) dest)) {
+                     ipv6_buf->destaddr.uint16[7] == (dest[1] << 8 | dest[0])) {
```

Comment:
@authmillenon for reference, could you point us to an example in the new
network stack in the master tree with proper usage of the network uint
types?
On Mar 28, 2015 5:42 PM, "Martine Lenders" <notifications@github.com> wrote:

> In sys/net/network_layer/sixlowpan/lowpan.c
> <https://github.com/RIOT-OS/RIOT/pull/2727#discussion_r27346310>:
>
> > @@ -1110,7 +1110,7 @@ uint8_t lowpan_iphc_encoding(int if_id, const uint8_t *dest, int dest_len,
> >              else if (dest_len == 2 &&
> >                       ipv6_buf->destaddr.uint32[2] == HTONL(0x000000ff) &&
> >                       ipv6_buf->destaddr.uint16[6] == HTONS(0xfe00) &&
> > -                     ipv6_buf->destaddr.uint16[7] == *((uint16_t *) dest)) {
> > +                     ipv6_buf->destaddr.uint16[7] == (dest[1] << 8 | dest[0])) {
>
> Using the old byteorder macros causes headaches for me, since I never know
> when anything is in what order (one reason why I use network_uintX_t in
> the new network stack very stringent and why the code of the old is so
> confusing and buggy ^^).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RIOT-OS/RIOT/pull/2727/files#r27346310>.
>